src/components/AppContent/AppContent.vue

```diff
@@ -120,6 +120,16 @@ import isMobile from '../../mixins/isMobile'
 
 const browserStorage = getBuilder('nextcloud').persist().build()
 
+let hasGlobalAppName = true
+let globalAppName
+try {
+	// This will throw a ReferenceError when the global variable is missing
+	// In that case we have to make the pane config key required
+	globalAppName = appName
+} catch (e) {
+	hasGlobalAppName = false
+}
```

Comment:
Honestly, this is even worse in terms of code clarity.

Either put that whole block as a service and import it or just remove the appName and make it required

Reply:
Please don't make it required, this will break all apps.